contracts/PoolTest.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "./Pool.sol";
+
+contract ReceiverWeightsTest {
+    using ReceiverWeightsImpl for mapping(address => ReceiverWeight);
+
+    mapping(address => ReceiverWeight) private receiverWeights;
+    ReceiverWeightIterated[] private receiverWeightsIterated;
+    int256 public receiverWeightsSumDelta;
+
+    struct ReceiverWeightIterated {
+        address receiver;
+        uint32 weight;
+    }
+
+    function setWeights(ReceiverWeightIterated[] calldata weights) external {
+        int256 weightsSumDelta = 0;
+        for (uint256 i = 0; i < weights.length; i++) {
+            weightsSumDelta += weights[i].weight;
+            weightsSumDelta -= receiverWeights.setWeight(
+                weights[i].receiver,
+                weights[i].weight
```

Comment:
It hurts my eyes, but `ReceiverWeightIterated calldata weight = weights[i];` does not compile. It's not possible to store a structure on the stack either. Is the only way to clean up the code here to copy the interesting item to memory? That would be so wasteful!

Reply:
Completely false, it just works. VSCode's in-editor error highlighter is wrong.